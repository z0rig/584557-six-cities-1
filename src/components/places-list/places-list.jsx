import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";

import PalceCard from "../place-card/place-card.jsx";

import {getCurrentCityOffers} from "../../reducer/offers/selectors";

import withActiveItem from "../../HOCs/with-active-item.jsx";

const PlacesList = (props) => {
  const {offers, activeItemId, changeActiveItemId} = props;

  return (
    <div className="cities__places-list places__list tabs__content">
      {offers ? offers.map((offer, ind) => {
        return (
          <PalceCard
            key={ind}
            offer={offer}
            activeItemId={activeItemId}
            itemClickHandler={changeActiveItemId}
          />
        );
      }) : null}
    </div>
  );
};

PlacesList.propTypes = {
  offers: PropTypes.arrayOf(PropTypes.object),
  activeItemId: PropTypes.string,
  changeActiveItemId: PropTypes.func
};

const mapStateToProps = (state) => {
  return {
    offers: getCurrentCityOffers(state)
  };
};

export {PlacesList};

export default connect(mapStateToProps)(withActiveItem(PlacesList));
