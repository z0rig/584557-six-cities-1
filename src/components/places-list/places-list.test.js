import React from "react";
import renderer from "react-test-renderer";
import {PlacesList} from "./places-list.jsx";

const mock = [
  {
    id: `asdf0`,
    isPremium: true,
    imgSrc: `img/apartment-01.jpg`,
    price: 12000,
    period: `day`,
    isFavorite: true,
    rating: 5,
    title: `Old palace`,
    housingType: `Palace`,
    coord: [52.3909553943508, 4.85309666406198]
  }
];

it(`PlacesList correctly renders`, () => {
  const tree = renderer.create(<PlacesList offers={mock} />).toJSON();
  expect(tree).toMatchSnapshot();
});
