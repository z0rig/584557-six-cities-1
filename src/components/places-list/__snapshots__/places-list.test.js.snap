// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesList correctly renders 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <article
    className="cities__place-card place-card"
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          onClick={[Function]}
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            12000
          </b>
          <span
            className="place-card__price-text"
          >
            / 
            day
          </span>
        </div>
        <button
          className="place-card__bookmark-button button place-card__bookmark-button--active"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "5%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
        >
          Old palace
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Palace
      </p>
    </div>
  </article>
</div>
`;
